### PR TITLE
Remove logic/operations from services #958

### PR DESCRIPTION
**Pull request by CaitlinOCallaghan**
- added error check when determining which service to call -> return promise with value 'null'
- remove logic/operations from service calls
- make service calls generic (manipulate id formatting in enrichment file, rather than in server-api)

### CODE REVIEW

File: src/client/features/enrichment-map/cy/index.js [hunk 1]
```diff
@@ -30,11 +30,19 @@ let bindEvents = cy => {
     let node = evt.target;
     let getEnrichmentTooltipData = () => {
       let id = node.data('id');
-      if(id.includes('GO')) return ServerAPI.getGoInformation(id);
-      else if(id.includes('REAC')) return ServerAPI.getReactomeInformation(id);
+      if(id.includes('GO')) return ServerAPI.getGoInformation( id.replace("GO:", "") );
```
Comment (jvwong) [inline src/client/features/enrichment-map/cy/index.js]: Safer if these were tests against a regex. I can't find any solid db id information for GO/Reactome, but seem to be similar to `/^GO:\d+$/` etc.

File: src/client/services/server-api/index.js [hunk 2]
```diff
@@ -27,23 +27,15 @@ const ServerAPI = {
   },
 
   getGoInformation(goID) {
-    return fetch(`https://www.ebi.ac.uk/QuickGO/services/ontology/go/search?query=GO%3A${goID.replace("GO:", "")}&limit=1&page=1`, {method: 'GET', timeout: 100})
+    return fetch(`https://www.ebi.ac.uk/QuickGO/services/ontology/go/search?query=GO%3A${goID}&limit=1&page=1`, {method: 'GET', timeout: 100})
```
Comment (jvwong) [inline src/client/services/server-api/index.js]: Rethrow any errors for the UI to deal with. See https://github.com/PathwayCommons/app-ui/pull/962#issuecomment-411498990

File: src/client/features/enrichment-map/cy/index.js [hunk 3]
```diff
@@ -30,11 +30,19 @@ let bindEvents = cy => {
...
+      else if(id.includes('REAC')) return ServerAPI.getReactomeInformation( id.replace("REAC:", "R-HSA-") );
+      else return Promise.resolve(null);
     };
 
-    getEnrichmentTooltipData().then( pathwayOverview => {
+    getEnrichmentTooltipData().then( result => {
```
Comment (jvwong) [inline src/client/features/enrichment-map/cy/index.js]: Need a `catch()` here and strategy to deal with it.

File: src/client/services/server-api/index.js (hunk 2 above)
Comment (maxkfranz) [reply src/client/services/server-api/index.js]: It's simpler to remove the catch altogether rather than rethrow within the catch.

Comment (jvwong) [reply src/client/services/server-api/index.js]: Remove `catch`

**Pull request merged**